libraries/standard/http/src/http_client.c

```diff
@@ -245,6 +245,986 @@ static int findHeaderValueParserCallback( http_parser * pHttpParser,
+    pParsingContext->pBufferCur += HTTP_HEADER_END_INDICATOR_LEN;
+
+    /* If headers existed, then pResponse->pHeaders was set during the first
+     * call to httpParserOnHeaderFieldCallback(). */
+    if( pResponse->pHeaders != NULL )
+    {
+        /* The start of the headers ALWAYS come before the the end of the headers. */
+        assert( ( const char * ) ( pResponse->pHeaders ) < pParsingContext->pBufferCur );
+
+        /* MISRA Rule 10.8 flags the following line for casting from a signed
+         * pointer difference to a size_t. This rule us suppressed because in
+         * in the previous statement it is asserted that the pointer difference
+         * will never be negative. */
+        /* coverity[misra_c_2012_rule_10_8_violation] */
+        pResponse->headersLen = ( size_t ) ( pParsingContext->pBufferCur - ( const char * ) ( pResponse->pHeaders ) );
+    }
+    else
+    {
+        pResponse->headersLen = 0u;
+    }
+
+    /* If the Content-Length header was found, then pHttpParser->content_length
+     * will not be equal to the maximum 64 bit integer. */
+
+    /* MISRA rule 14.3 flags the following line as invaraint. http-parser library
+     * sets pHttpParser->content_length intentionally to the maximum 64 bit
+     * integer to note a non-existent Content-Length header. */
+    /* coverity[misra_c_2012_rule_14_3_violation] */
+    if( pHttpParser->content_length != ( ( uint64_t ) -1 ) )
+    {
+        pParsingContext->pResponse->contentLength = ( size_t ) ( pHttpParser->content_length );
+    }
+    else
+    {
+        pParsingContext->pResponse->contentLength = 0u;
+    }
+
+    /* If the Connection: close header was found this flag will be set. */
+
+    /* The MISRA directive 4.6 violation is for using primitive type unsigned
+     * int, instead of a typedef which denotes the signedness and size of the
+     * type. This violation is suppressed because http-parser defines these
+     * variables as an unsigned int. */
+    /* coverity[misra_c_2012_directive_4_6_violation] */
+    if( ( pHttpParser->flags & ( unsigned int ) ( F_CONNECTION_CLOSE ) ) != 0u )
+    {
+        pResponse->flags |= HTTP_RESPONSE_CONNECTION_CLOSE_FLAG;
+    }
+
+    /* If the Connection: keep-alive header was found this flag will be set. */
+
+    /* The MISRA directive 4.6 violation is for using primitive type unsigned
+     * int, instead of a typedef which denotes the signedness and size of the
+     * type. This violation is suppressed because http-parser defines these
+     * variables as an unsigned int. */
+    /* coverity[misra_c_2012_directive_4_6_violation] */
+    if( ( pHttpParser->flags & ( unsigned int ) ( F_CONNECTION_KEEP_ALIVE ) ) != 0u )
+    {
+        pResponse->flags |= HTTP_RESPONSE_CONNECTION_KEEP_ALIVE_FLAG;
+    }
+
+    /* http_parser_execute() requires that callback implementations must
+     * manually stop the parsing on headers complete, if response is to a HEAD
```

Comment:
'manually' may not be the right word here.

Reply:
```
    /* http_parser_execute() requires that callback implementations must
     * indicate that parsing stops on headers complete
```